app/src/main/java/com/example/trainboard/MainActivity.kt

```diff
@@ -1,47 +1,160 @@
+        Row (){
+            SimpleExposedDropdown("From", selectedStartStation, selectedEndStation)
+            SimpleExposedDropdown("To", selectedEndStation, selectedStartStation)
+
+        }
+        ButtonToLNER(selectedStartStation, selectedEndStation)
+    }
+
+}
 
+
+fun generateUrl(selectedStartStation: MutableState<String>, selectedEndStation: MutableState<String>): String{
+    val codeDictionary = mutableMapOf("London" to "KGX", "Edinburgh" to "EDB", "Oxford" to "OXF", "Bristol" to "BRI", "Liverpool" to "LVC")
+    return "https://www.lner.co.uk/travel-information/travelling-now/live-train-times/depart/${codeDictionary[selectedStartStation.value]}/${codeDictionary[selectedEndStation.value]}/#tab_livedepartures"
+}
 @Composable
-fun Greeting(name: String, modifier: Modifier = Modifier) {
-    Text(
-        text = "Hello $name!",
-        modifier = modifier
+fun ButtonToLNER (selectedStartStation: MutableState<String>, selectedEndStation: MutableState<String>)
+{   val context = LocalContext.current
+    val buttonText = remember { mutableStateOf("Find route") }
+    val url = generateUrl(selectedStartStation, selectedEndStation)
+    Button(
+
+        onClick = {
+
+            if (selectedStartStation.value != "" && selectedEndStation.value != "")
+            {
+                val intent = Intent(Intent.ACTION_VIEW, url.toUri())
+
+                context.startActivity(intent)
+            }
+            else {
+
+                buttonText.value = "Stations not chosen"
+            }
+
+        },
+        colors = ButtonDefaults.buttonColors(
+            containerColor = Color.White,
+            contentColor = Color.Black
+        )
     )
+    {
+        Text(buttonText.value)
+    }
+
+
 }
 
-@Preview(showBackground = true)
+@OptIn(ExperimentalMaterial3Api::class)
 @Composable
-fun GreetingPreview() {
-    TrainBoardTheme {
-        Greeting("Android")
+fun SimpleExposedDropdown(name: String, selectedOptionText: MutableState<String>,selectedOtherOptionText: MutableState<String>) {
+    val options = listOf("London", "Edinburgh", "Oxford", "Liverpool", "Bristol")
+    var expanded by remember { mutableStateOf(false) }
+    ExposedDropdownMenuBox(
+        expanded = expanded,
+        onExpandedChange = { expanded = !expanded }
+    ) {
+        TextField(
+            value = selectedOptionText.value,
+            onValueChange = {},
+            readOnly = true,
+            label = { Text(name) },
+            trailingIcon = {
+                ExposedDropdownMenuDefaults.TrailingIcon(expanded)
+            },
+            modifier = Modifier.menuAnchor().clickable { expanded = !expanded }.padding(5.dp).width(150.dp)
+        )
+
+        ExposedDropdownMenu(
+            expanded = expanded,
+            onDismissRequest = { expanded = false }
+        ) {
+            options.forEach { selectionOption ->
+                if (selectionOption != selectedOtherOptionText.value){
```

Comment:
```suggestion
               options.filter { it != selectedOtherOptionText.value }.forEach { selectionOption ->
    DropdownMenuItem(
```

to avoid the nested for -> if loop.